src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstanceEventsController.cs

```diff
@@ -0,0 +1,177 @@
+        public InstanceEventsController(IInstanceEventRepository instanceEventRepository)
+        {
+            _repository = instanceEventRepository;
+        }
+
+        /// <summary>
+        /// Inserts new instance event into the instanceEvent collection.
+        /// </summary>
+        /// <param name="instanceEvent">Id of instance to retrieve events for. </param>
+        /// <returns>The stored instance event.</returns>
+        /// POST storage/api/v1/instanceevents/
+        [HttpPost]
+        public async Task<ActionResult> Post([FromBody] InstanceEvent instanceEvent)
+        {
+            if (instanceEvent == null)
+            {
+                return BadRequest("Missing parameter values: instance event must exist and instanceId must be set");
+            }
+
+            instanceEvent.CreatedDateTime = DateTime.UtcNow;
+
+            InstanceEvent result = await _repository.InsertInstanceEvent(instanceEvent);
+            if (result == null)
+            {
+                return BadRequest("Unable to write new instance event to database");
+            }
+
+            return Ok(result);
+        }
+
+        /// <summary>
+        /// Retrieves all instance events related to given instance id from instanceEvent collection.
+        /// </summary>
+        /// <param name="instanceId">Id of instance to retrieve events for. </param>
+        /// <returns>List of instance events.</returns>
+        /// GET  storage/api/v1/instanceevents/?instanceid={instanceId}
+        [HttpGet]
+        public async Task<ActionResult> GetAllInstanceEvents(string instanceId)
+        {
+            if (!string.IsNullOrEmpty(instanceId))
+            {
+                List<InstanceEvent> result = await _repository.ListAllInstanceEvents(instanceId);
+                if (result == null || result.Count == 0)
+                {
+                    return NotFound($"Did not find any instance events for instanceId={instanceId}");
+                }
+
+                return Ok(result);
+            }
+
+            return BadRequest("Unable to perform query.");
+        }
+
+        /// <summary>
+        /// Retrieves all instance events related to given instance id and listed event types from instanceEvent collection.
+        /// </summary>
+        /// <param name="instanceId">Id of instance to retrieve events for.</param>
+        /// <param name="eventTypes">List of event types to filter the events by./param>
+        /// <returns>List of intance events.</returns>
+        /// GET  storage/api/v1/instanceevents/GetByInstanceEventType?instanceId={instanceId}&eventTypes={eventType}&eventTypes={eventType}
+        [HttpGet]
+        [Route("GetByInstanceEventType")]
+        public async Task<ActionResult> GetInstanceEventsEventTypes(string instanceId, List<string> eventTypes)
+        {
+            // or array of types is empty
+            if (string.IsNullOrEmpty(instanceId) || eventTypes.Count == 0 || eventTypes == null)
```

Comment:
Should check eventTypes == null before eventTypes.Count. Otherwise nullpointer exception can occur.

Reply:
This method was removed, but added a null check inn all other places where the count is checked.